proposal.rst

```diff
@@ -237,7 +247,51 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+* Pre-acceptance Period from Present to May 5th
+	This one month period can be utilized to work on issues related to omega so that I can get familiar to the codebase and also to gain some familiarity to Linux system programming.
+	
+* Community Bonding Period from May 6th to May 26th
+	Understand the usage of the shared libraries to be used to replace external filter programs by reading the documentation and discussing it with mentors and continue working on issues related to omega.
+
+					 ----- Coding officially begins -----
+
+* Week 1 from May 27th to June 2nd
+	Testing Omega before adding any library to the project and check how it runs with the verbose mode to get the grasp of the sequence of implementation of current source code. Continue search and understanding the usage of libraries. Discuss the outcomes with mentors.
+
+* Week 2 and 3 from June 3rd to June 16th
+	Implementing the libarchive library for reading zip file formats. Testing and documentation along. (This phase may take a longer time than expected as this is the beginning of coding)
+
+* Week 4 from June 17th to June 23rd
+	Implementing the libpoppler library for reading zip file formats. Testing and documentation along. Getting libarchive and libpoppler to work before Phase I evaluation will be my main goal.
+
+					 ----- Phase 1 Evaluation -----
+
+* Week 5 and 6 from June 24th to July 7th
+	Implementing the DjVuLibre library for reading DjVu format. Testing and documentation along.
+
+* Week 7 from July 8th to July 14th
+	Although the libarchive would be able to read the AbiWord files, the extent of AbiWord is large. Hence, I would like to implement the library libabw in case the library crashes. This week can be devoted to it. Testing and documentation all along.
```

Comment:
"in case the library crashes" — in case libarchive crashes you'd use libabw? Or…I'm not sure I understand what you're proposing here.

Reply:
libarchive handles many types of formats but libabw is specific to AbiWord. Hence, the first thought that came to my mind was that having a separate library for AbiWord documents would be better (I assumed that as libabw is made targetting only AbiWord documents, it would be less likely for it to crash than libarchive) if libarchive crashes for an AbiWord file, we can use libabw.

Reply:
I don't think libarchive can read AbiWord documents - https://github.com/libarchive/libarchive/wiki/LibarchiveFormats certainly doesn't list it as a format, and it isn't an archive or compression format, which is what libarchive's target area is (a `.abw` file is actually an XML document).

But anyway, AbiWord is a format already handled without needing any external filter program (via the XmlParser class).  It's also not a widely used format any more - e.g. https://qa.debian.org/popcon-graph.php?packages=abiword shows the number of reported installs by Debian machines running the optional popcon reporting package - the peak was about 6 years ago and it's tailing off significantly now.  I think most people in the FLOSS world who want a word processor use LibreOffice and OpenDocument files currently.  We still want to support AbiWord so people can index existing documents, but it's not a key format.  So I don't think it's useful to put effort into supporting it via a specific or general third-party library.

More generally, it seems dubious to assume that a more general purpose library would be more crash-prone than a more specific purpose library for a particular file format.  I can see your logic, but really it depends so much on how well maintained and tested each library is - for example, a general-purpose format library with a good automated test suite that's used in a widely used application is likely to more solid than a specific format library which has no automated testsuite and isn't used much.

Reply:
Thanks, I understood. Then we can devote that time for adding support for other formats. One question. How can I set priority for the formats to be added so that I can cover more widely used formats in this project?

Reply:
It's hard to really know what formats our users (and potential users) are actually caring most about, but some are fairly clearly minority formats and others very mainstream.  E.g. djvu is clearly not widely used, but PDF definitely is.

I'd also tend to favour putting effort in to supporting an open format over a proprietary one when the two have similar popularity.

Reply:
So should I skip DjVu and try to cover the formats from Document Liberation project?

Reply:
Probably the latter are more common than the former.

I'm not sure I'd skip anything though, but rather move it later or last (you could make it explicitly a stretch goal).

The first format (whatever it is) will probably be most work as the framework for these plugins needs creating, and after that I'd expect each additional format would be increasingly quicker.  A new format might reveal some shortcoming in the framework which needs adjusting, but that's going to become less likely as more formats get added.  So you could well find you fly through the last few formats.

Reply:
Yes, that makes sense. I will shift the DjVu format to the end.